app.js

```diff
@@ -0,0 +1,140 @@
+function Menu(name) {
+    this.name = name;
+}
+
+Menu.prototype.calculatePrice = function() {
+    return this.name.price;
+}
+
+Menu.prototype.calculateCalories = function() {
+    return this.name.calories;
+}
+
+function Hamburger(name, ...stuffing) { 
+    Menu.call(this, name);
+    this.stuffing = stuffing;
+} 
+
+Hamburger.SIZE_SMALL = { type: 'small', price: 50, calories: 20 };
+Hamburger.SIZE_LARGE = { type: 'large', price: 100, calories: 40 };
+Hamburger.STUFFING_CHEESE = { type: 'cheese', price: 10, calories: 20 };
+Hamburger.STUFFING_SALAD = { type: 'salad', price: 20, calories: 5 };
+Hamburger.STUFFING_POTATO = { type: 'potato', price: 15, calories: 10 };
+
+Hamburger.prototype = Object.create(Menu.prototype);
+
+Hamburger.prototype.calculatePrice = function () {
+    let count_stuff = 0;
+
+    for (item in this.stuffing) {
+        count_stuff += this.stuffing[item].price;
+    };
+
+    return this.name.price+count_stuff;
+}
+
+Hamburger.prototype.calculateCalories = function () {
+    let hamburger_calories = 0;
+
+    for (item in this.stuffing) {
+        hamburger_calories += this.stuffing[item].calories;
+    }
+
+    return hamburger_calories+this.name.calories;
+}
+
+function Salad(name, weight) {
+    Menu.call(this, name);
+    this.weight = weight;
+}
+
+Salad.CAESAR = {type: 'caesar', price: 100, calories: 20};
+Salad.OLIVIE = {type: 'olivie', price: 50, calories: 80};
+
+Salad.prototype = Object.create(Menu.prototype);
+
+Salad.prototype.calculatePrice = function() {
+    return this.weight*this.name.price*0.01;
```

Comment:
0.01 - is a magic number. Please add a constant at the top of the file 
const CONSTANT_NAME = value;